### PR TITLE
ScrollSpy requires non-body elements to have height and overflow…

### LINKED ISSUE #16441: ScrollSpy works only on the body tag?
I've got an example here: http://jsfiddle.net/03sv47av/

As is, scrollspy will set up, but will not react to scrolling. But if you cut all the attributes from the first div and paste them to the body tag and re-run the example, scrollspy works perfectly.  

You'll need to inspect the nav elements with your browser dev tools. When it works, the active classes will be dynamically applied as you scroll. 

I've reread the docs about a zillion times now, and I can't seem to pinpoint where I'm going wrong.

### PR DESCRIPTION
**Pull request by kkirsche**
Fixes #16441

### CODE REVIEW

**Comment by mdo** [comment]
The docs change mentions height and overflow, but your description says height and padding.

**Comment by kkirsche** [comment]
Oops. Typo on my end. Fixing that now and will re-push

**Comment by kkirsche** [comment]
@mdo content was right, just not the title of the PR :+1:

**Comment by XhmikosR** [comment]
@kkirsche: no need to include close blah blah in the title. Also, fixes works too.

**Comment by kkirsche** [comment]
Thanks @XhmikosR I'll be better about that in future PRs

**Comment by kkirsche** [comment]
Wanted to follow up on this and see if it was missing anything

**Comment by cvrebert** [comment]
CC: @fat

**Comment by kkirsche** [comment]
Wanted to follow up on this and see if it was missing anything

File: docs/_includes/js/scrollspy.html [hunk 1]
```diff
@@ -65,7 +65,7 @@ <h2 id="scrollspy-usage">Usage</h2>
   </div>
 
   <h3>Requires relative positioning</h3>
-  <p>No matter the implementation method, scrollspy requires the use of <code>position: relative;</code> on the element you're spying on. In most cases this is the <code>&lt;body&gt;</code>.</p>
+  <p>No matter the implementation method, scrollspy requires the use of <code>position: relative;</code> on the element you're spying on. In most cases this is the <code>&lt;body&gt;</code>. In some cases though, you may use another element, this element will require a valid height and overflow to be set on the element being spied on.</p>
```
Comment (cvrebert) [inline docs/_includes/js/scrollspy.html]: Punctuation needs a tweak and there seems to be an extra "element". Correction:
> [...] another element; this ~~element~~ will require [...]

**Comment by cvrebert** [comment]
@kkirsche Mostly I'm just waiting for @fat to review this. He tends to be busy with other things.

**Comment by kkirsche** [comment]
Ah ok. Sorry. I'll reduce how often I follow up on this one so as not to drown anyone with messages but still keep it in view

File: docs/_includes/js/scrollspy.html (hunk 1 above)
Comment (kkirsche) [inline docs/_includes/js/scrollspy.html]: Adjusted and simplified the grammar of the sentance. I think it should be a bit clearer now. Let me know though if you have anymore feedback :+1:

**Comment by kkirsche** [comment]
Wanted to follow up on this and see if it was missing anything?

**Comment by cvrebert** [comment]
As before, still just waiting on @fat.

**Comment by kkirsche** [comment]
Ok, when would you like me to follow up on this topic? I read your comment to mean that I have followed to soon, and I apologize if you feel I am being annoying here. I personally feel 10 days is a sufficient time for a response simply stating this is low on his priority list and a rough idea of his timeline even if it's just he'll take a look at end of Q2, as to avoid nagging you guys. This could even be a core team member stating "we aren't sure when he'll review it but try following up in X days" so we have an idea what your expectations are to ensure us, contributors, understand these things. Without a team member setting the expectations its left to people like me who may have much stricter expectations of others than some may have. 

I can certainly understand others are busy and I don't expect nor do I mean to imply that the actual review should always happen quickly or could always happen quickly but personally it can be frustrating and at times off putting to not get real feedback or timelines from a project and instead receive an open ended "just wait…". This leads to users such as me frustrating the core team because we would like more information when our expectations of timelines don't match up. 

I hope you and the core team can understand this point of view from a contributor perspective and do not take this as an insult to you or the team. You all do a great job and I appreciate the time and assistance that you and the team provide to me and other contributors. I understand this is free and volunteer, and at no time should something like this become a job, which I'm sure it starts to feel like. I greatly appreciate the time and work done here and appreciate your understanding. I hope it will be possible to set clearer expectations in future situations like this to avoid unnecessary noise and frustration for any / all involved.

**Comment by mdo** [comment]
Definitely our bad @kkirsche. We need to grow the team a bit more I think to help triage and communicate things more. Jacob and I aren't always readily available—even across a couple weeks' span with work and other projects—but that shouldn't hold folks up like this.

This one was on me though—I forgot I left feedback and didn't check back in after you pushed some updates to resolve my feedback. Thanks for doing so!

I've got one more copy edit I'd like to make and then I'm merge this right away:

> When scrollspying on elements other than the `<body>`, be sure to have a <code>height</code> set and <code>overflow-y: scroll;</code> applied.

I think that should be sufficient enough, unless anyone has more feedback on that.

**Comment by kkirsche** [comment]
No worries @mdo. I fully understand that it can be difficult to find the time as I know all the members on the core team, as well as contributors outside of the project, are very busy with a number of projects (as I try to take a look through Github public repo's), work and their own lives. If there is anything I can do to help out, feel free to let me know. I hope my original post came across correctly and didn't come across as rude or insulting.

I've pushed the updated fix with the copy change you mentioned. Let me know if there are any other changes you'd like to see and I'll make those ASAP for you guys.

Thank you for your and the core team's time and assistance.

**Pull request merged**